### ISSUE TITLE
Kong 3.0.0 fail to start after migration due to incorrect routes value

### ISSUE THREAD

**Issue by cypx** [Kong/kong#9450]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

Kong 3.0

### Current Behavior

Kong fail to start after upgrade from 2.8.1 to 3.0.0.

### Expected Behavior

Successful start.

### Steps To Reproduce

```markdown
Upgrade Kong from 2.8.1 to 3.0.0, launch `migrations up` and `migrations finish`, try to start.

Start fail and log report an error on **atc_compat.lua** about existing route


Sep 20 14:46:43 localhost.localdomain systemd[1]: Starting kong service...
-- Subject: Unit kong.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit kong.service has begun starting up.
Sep 20 14:46:44 localhost.localdomain kong[20157]: Error: /usr/local/share/lua/5.1/kong/cmd/start.lua:101: nginx: [error] init_by_lua error: /usr/local/share/lua/5.1/kong/router/atc_compat.lua:411:  --> 1:2
Sep 20 14:46:44 localhost.localdomain kong[20157]: |
Sep 20 14:46:44 localhost.localdomain kong[20157]: 1 | () && (http.path ^= "/analytics")
Sep 20 14:46:44 localhost.localdomain kong[20157]: |  ^---
Sep 20 14:46:44 localhost.localdomain kong[20157]: |
Sep 20 14:46:44 localhost.localdomain kong[20157]: = expected term
Sep 20 14:46:44 localhost.localdomain kong[20157]: stack traceback:
Sep 20 14:46:44 localhost.localdomain kong[20157]: [C]: in function 'assert'
Sep 20 14:46:44 localhost.localdomain kong[20157]: /usr/local/share/lua/5.1/kong/router/atc_compat.lua:411: in function 'add_atc_matcher'
Sep 20 14:46:44 localhost.localdomain kong[20157]: /usr/local/share/lua/5.1/kong/router/atc_compat.lua:436: in function 'new_from_scratch'
Sep 20 14:46:44 localhost.localdomain kong[20157]: /usr/local/share/lua/5.1/kong/router/atc_compat.lua:534: in function 'new'
Sep 20 14:46:44 localhost.localdomain kong[20157]: /usr/local/share/lua/5.1/kong/runloop/handler.lua:802: in function 'build_router'
Sep 20 14:46:44 localhost.localdomain kong[20157]: /usr/local/share/lua/5.1/kong/init.lua:582: in function 'init'
Sep 20 14:46:44 localhost.localdomain kong[20157]: init_by_lua:3: in main chunk
Sep 20 14:46:44 localhost.localdomain kong[20157]: Run with --v (verbose) or --vv (debug) for more details
Sep 20 14:46:44 localhost.localdomain systemd[1]: kong.service: control process exited, code=exited status=1
Sep 20 14:46:44 localhost.localdomain systemd[1]: Failed to start kong service.
-- Subject: Unit kong.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit kong.service has failed.
-- 
-- The result is failed.
Sep 20 14:46:44 localhost.localdomain systemd[1]: Unit kong.service entered failed state.
Sep 20 14:46:44 localhost.localdomain systemd[1]: kong.service failed.
```

### Anything else?

Fixed by update all rows with empty array values on methods and hosts 

```sql
UPDATE public.routes SET methods=NULL WHERE methods='{}';
UPDATE public.routes SET hosts=NULL WHERE hosts='{}';
```

**Comment by hbagdi** [comment]
@dndx Potentially a candidate for 3.0.1?

**Comment by dndx** [comment]
@hbagdi This got on our radar already, and is high priority to fix.

**Comment by dndx** [comment]
This has been addressed in https://github.com/Kong/kong/pull/9451, and will be released in the upcoming 3.0.1 release.
